cmd/cert-checker/main.go

```diff
@@ -355,46 +359,74 @@ func main() {
 	cmd.FailOnError(err, "Failed to dial syslog")
+
 	logger, err := blog.New(syslogger, 0, 0)
 	cmd.FailOnError(err, "Failed to construct logger")
+
 	err = blog.Set(logger)
 	cmd.FailOnError(err, "Failed to set audit logger")
 
-	avps := make(map[uint]bool)
-	if len(config.CertChecker.AcceptableValidityPeriods) == 0 {
-		// For backwards compatibility, assume only a single valid validity period
-		// of exactly 90 days if none is configured.
-		avps[uint((time.Hour * 24 * 90).Seconds())] = true
+	// TODO(#5581): This check can be removed once staging and production configs
+	// use `acceptableValidityDurations`.
+	if len(config.CertChecker.AcceptableValidityDurations) > 0 && len(config.CertChecker.AcceptableValidityPeriods) > 0 {
+		cmd.Fail("Config specifies both 'acceptableValidityDurations' and 'acceptableValidityPeriods'")
+	}
+
+	acceptableValidityDurations := make(map[time.Duration]bool)
+	if len(config.CertChecker.AcceptableValidityDurations) == 0 && len(config.CertChecker.AcceptableValidityPeriods) == 0 {
+		// For backwards compatibility, assume only a single valid validity
+		// period of exactly 90 days if none is configured.
+		ninetyDays := (time.Hour * 24) * 90
+		acceptableValidityDurations[ninetyDays] = true
 	} else {
-		for _, period := range config.CertChecker.AcceptableValidityPeriods {
-			avps[period] = true
+		if len(config.CertChecker.AcceptableValidityDurations) > 0 {
+			for _, duration := range config.CertChecker.AcceptableValidityDurations {
+				acceptableValidityDurations[duration.Duration] = true
+			}
+		}
+		if len(config.CertChecker.AcceptableValidityPeriods) > 0 {
+			// TODO(#5581): This conditional is deprecated and can be removed once
+			// staging and production configs use `acceptableValidityDurations`.
+			for _, period := range config.CertChecker.AcceptableValidityPeriods {
+				duration, err := time.ParseDuration(fmt.Sprintf("%ds", period))
+				if err != nil {
+					cmd.FailOnError(err, "Failed to marshal period to time.Duration")
+				}
+				acceptableValidityDurations[duration] = true
+			}
 		}
 	}
 
-	// Validate PA config and set defaults if needed
+	// Validate PA config and set defaults if needed.
 	cmd.FailOnError(config.PA.CheckChallenges(), "Invalid PA configuration")
 
 	saDbURL, err := config.CertChecker.DB.URL()
 	cmd.FailOnError(err, "Couldn't load DB URL")
+
 	dbSettings := sa.DbSettings{
 		MaxOpenConns:    config.CertChecker.DB.MaxOpenConns,
 		MaxIdleConns:    config.CertChecker.DB.MaxIdleConns,
 		ConnMaxLifetime: config.CertChecker.DB.ConnMaxLifetime.Duration,
 		ConnMaxIdleTime: config.CertChecker.DB.ConnMaxIdleTime.Duration,
 	}
+
+	conf, err := mysql.ParseDSN(saDbURL)
+	cmd.FailOnError(err, "Couldn't parse DB URL as DSN")
+
+	// Set transaction isolation level to READ UNCOMMITTED. This trades
+	// consistency for performance.
+	if len(conf.Params) == 0 {
+		conf.Params = make(map[string]string)
+	}
+	conf.Params["tx_isolation"] = "'READ-UNCOMMITTED'"
+	saDbURL = conf.FormatDSN()
```

Comment:
Feels a bit odd for this transaction isolation level change to be included in the same PR; these are separate concerns which (in the worst case) we would want to revert+rollback separately.